src/yaml/schema/core.cr

```diff
@@ -84,6 +84,7 @@ module YAML::Schema::Core
       value = parse_float?(string)
       return value || string
     when .starts_with?('0')
+      return 0_i64 if string.size == 1
```

Comment:
Maybe `when '0'` as the first case branch might be a bit more obvious?

Reply:
I thought about that, too. But this way it's more closely expressing the reason why this special case is needed. It's not a big difference, so I'll be happy to change this if this is preferred.